internals/overlord/checkstate/manager.go

```diff
@@ -38,8 +37,7 @@ const (
 
 // CheckManager starts and manages the health checks.
 type CheckManager struct {
-	state      *state.State
-	ensureDone atomic.Bool
```

Comment:
This logic, which would also have been needed in derived project managers, can now safely be removed. The race is resolved in the overlord.